docs: how to check resource status (#535)
apache/apisix-ingress-controller: docs/en/latest/practices/check-crd-status.md

---
title: How to quickly check the synchronization status of CRD
---

<!--
#
# Licensed to the Apache Software Foundation (ASF) under one or more
# contributor license agreements.  See the NOTICE file distributed with
# this work for additional information regarding copyright ownership.
# The ASF licenses this file to You under the Apache License, Version 2.0
# (the "License"); you may not use this file except in compliance with
# the License.  You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
-->

> When using the APISIX ingress declarative configuration, often use the `kubectl apply` command. If the configuration is verified by the schema and validation webhook, the configuration will be accepted by Kubernetes.

Reviewer: ```suggestion
When using the APISIX ingress declarative configuration, often use the `kubectl apply` command. Only if the configuration was verified by its [Open API V3 Schema definition](https://swagger.io/specification/) and its validation webhooks (if any), can the configuration be accepted by Kubernetes.
```

Author: OK

> When the Ingress Controller watches the resource change, the logic unit of the Ingress Controller has just started to work. 

Reviewer: Use term Apache APISIX Ingress Controller.

In various operations of the Apache APISIX Ingress Controller, object conversion and more verification will be performed. 

> When the verification fails, the Apache APISIX Ingress Controller will throw an error message and will continue to retry 

Reviewer 2: ```suggestion
When the verification fails, the Apache APISIX Ingress Controller will log an error message and will continue to retry 
```

until the declared state is successfully synchronized to APISIX.

Therefore, after the declarative configuration is accepted by Kubernetes, it does not mean that the configuration is synchronized to APISIX.

> In this practice, we will show how to  check the status of CRD.

Reviewer: ```suggestion
In this practice, we will show how to check the status of CRD.
```

## Prerequisites

> - an available Kubernetes cluster

Reviewer: We may add some version hints.

- an available Apache APISIX and Apache APISIX Ingress Controller installation

## Take ApisixRoute resource as an example 

### deploy and check ApisixRoute resource

1. first deploy an ApisixRoute resource

e.g.
   
```yaml
kubectl apply -f - <<EOF
apiVersion: apisix.apache.org/v2alpha1
kind: ApisixRoute
metadata:
  name: httpbin-route

>   namespace: test

Reviewer 2: If namespace is specified here, you need to create it first.

Author: Namespapce removed.
